dummyserver/testcase.py

```diff
@@ -109,6 +113,10 @@ def setUpClass(cls):
         cls.proxy_server, cls.proxy_port = run_tornado_app(
             app, cls.io_loop, None, 'http', cls.proxy_host)
 
+        app = web.Application([(r'.*', ProxyHandler)])
```

Comment:
Is there a reason why you are running two proxies in one test case? Would make more sense if you make a separate class as HTTPS proxy test case.